### PR TITLE
treewide: vendorSha256 → vendorHash

### PR DESCRIPTION
**Pull request by fabianhjr**
## Description of changes

From this review: https://github.com/NixOS/nixpkgs/pull/254853#pullrequestreview-1623507375

Should be a 0 rebuild change and help along the migration from sha256 to hash

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: doc/languages-frameworks/go.section.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ In the following is an example expression using `buildGoModule`, the following a
 
   To obtain the actual hash, set `vendorHash = lib.fakeSha256;` and run the build ([more details here](#sec-source-hashes)).
 - `proxyVendor`: Fetches (go mod download) and proxies the vendor directory. This is useful if your code depends on c code and go mod tidy does not include the needed sources to build or if any dependency has case-insensitive conflicts which will produce platform-dependent `vendorHash` checksums.
-- `modPostBuild`: Shell commands to run after the build of the goModules executes `go mod vendor`, and before calculating fixed output derivation's `vendorHash` (or `vendorSha256`). Note that if you change this attribute, you need to update `vendorHash` (or `vendorSha256`) attribute.
+- `modPostBuild`: Shell commands to run after the build of the goModules executes `go mod vendor`, and before calculating fixed output derivation's `vendorHash` (or `vendorHash`). Note that if you change this attribute, you need to update `vendorHash` (or `vendorHash`) attribute.
```
Comment (fabianhjr) [inline doc/languages-frameworks/go.section.md:23]: ops messed this one up

File: pkgs/applications/misc/archiver/default.nix [hunk 2]
```diff
@@ -14,7 +14,7 @@ buildGoModule rec {
     sha256 = "1py186hfy4p69wghqmbsyi1r3xvw1nyl55pz8f97a5qhmwxb3mwp";
   };
 
-  vendorSha256 = "1y4v95z1ga111g3kdv5wvyikwifl25f36firf1i916rxli6f6g5i";
+  vendorHash = "1y4v95z1ga111g3kdv5wvyikwifl25f36firf1i916rxli6f6g5i";
```
Comment (dotlambda) [inline pkgs/applications/misc/archiver/default.nix:17]: This won't work.

Comment (fabianhjr) [reply pkgs/applications/misc/archiver/default.nix:17]: Managed to corrupt my .git, though detected all this cases with:

`git show HEAD | rg vendorHash | rg -v "(null|sha256)"`

**Comment by fabianhjr** [comment]
Rerunning the rewrite with:

`find pkgs/ -type f -exec sed -i 's/vendorSha256 = "sha256/vendorHash = "sha256/' {};`

**Comment by fabianhjr** [comment]
On x86_64-linux nixpkgs-review: Nothing to be built.

:3

**Comment by zowoq** [comment]
@ofborg eval

**Comment by zowoq** [comment]
`ofborg` is adding the `clean-up` label because this PR is removing a package, that needs to be addressed.

Note that `ofborg` will not remove the `clean-up` label automatically, need to remove it manually and make sure `ofborg` doesn't add it again.

**Comment by fabianhjr** [comment]
> because this PR is removing a package (or packages)

This is quite unexpected, even double checking the diff:

```
 pkgs/applications/editors/gophernotes/default.nix                               |  2 +-
 pkgs/applications/editors/hecate/default.nix                                    |  2 +-
 pkgs/applications/graphics/ascii-image-converter/default.nix                    |  2 +-
 pkgs/applications/misc/darkman/default.nix                                      |  2 +-
 pkgs/applications/misc/expenses/default.nix                                     |  2 +-
 pkgs/applications/misc/gmnitohtml/default.nix                                   |  2 +-
 pkgs/applications/misc/go-jira/default.nix                                      |  2 +-
 pkgs/applications/misc/go-thumbnailer/default.nix                               |  2 +-
 pkgs/applications/misc/gsctl/default.nix                                        |  2 +-
 pkgs/applications/misc/kiln/default.nix                                         |  2 +-
 pkgs/applications/misc/kratos/default.nix                                       |  2 +-
 pkgs/applications/misc/lemonade/default.nix                                     |  2 +-
 pkgs/applications/misc/mop/default.nix                                          |  2 +-
 pkgs/applications/misc/nwg-menu/default.nix                                     |  2 +-
 pkgs/applications/misc/openring/default.nix                                     |  2 +-
 pkgs/applications/misc/remarkable/rmapi/default.nix                             |  2 +-
 pkgs/applications/misc/senv/default.nix                                         |  2 +-
 pkgs/applications/misc/slides/default.nix                                       |  2 +-
 pkgs/applications/misc/sqls/default.nix                                         |  2 +-
 pkgs/applications/misc/tasktimer/default.nix                                    |  2 +-
 pkgs/applications/misc/tsukae/default.nix                                       |  2 +-
 pkgs/applications/misc/writefreely/default.nix                                  |  2 +-
 pkgs/applications/misc/ydict/default.nix                                        |  2 +-
 pkgs/applications/networking/bee/bee.nix                                        |  8 ++++----
 pkgs/applications/networking/browsers/amfora/default.nix                        |  2 +-
 pkgs/applications/networking/cluster/assign-lb-ip/default.nix                   |  2 +-
 pkgs/applications/networking/cluster/atmos/default.nix                          |  2 +-
 pkgs/applications/networking/cluster/cmctl/default.nix                          |  2 +-
 pkgs/applications/networking/cluster/cni/default.nix                            |  2 +-
 pkgs/applications/networking/cluster/fetchit/default.nix                        |  2 +-
 pkgs/applications/networking/cluster/fluxcd/default.nix                         |  2 +-
 pkgs/applications/networking/cluster/fluxctl/default.nix                        |  2 +-
 pkgs/applications/networking/cluster/hashi-up/default.nix                       |  2 +-
 pkgs/applications/networking/cluster/jx/default.nix                             |  2 +-
 pkgs/applications/networking/cluster/k0sctl/default.nix                         |  2 +-
 pkgs/applications/networking/cluster/kconf/default.nix                          |  2 +-
 pkgs/applications/networking/cluster/kubecolor/default.nix                      |  2 +-
 pkgs/applications/networking/cluster/kubectl-doctor/default.nix                 |  2 +-
 pkgs/applications/networking/cluster/kubectl-klock/default.nix                  |  2 +-
 pkgs/applications/networking/cluster/kubectl-ktop/default.nix                   |  2 +-
 pkgs/applications/networking/cluster/kubectl-tree/default.nix                   |  2 +-
 pkgs/applications/networking/cluster/kubent/default.nix                         |  2 +-
 pkgs/applications/networking/cluster/kubeval/default.nix                        |  2 +-
 pkgs/applications/networking/cluster/levant/default.nix                         |  2 +-
 pkgs/applications/networking/cluster/linkerd/default.nix                        |  2 +-
 pkgs/applications/networking/cluster/linkerd/edge.nix                           |  2 +-
 pkgs/applications/networking/cluster/linkerd/generic.nix                        |  4 ++--
 pkgs/applications/networking/cluster/nomad-autoscaler/default.nix               |  2 +-
 pkgs/applications/networking/cluster/nomad-pack/default.nix                     |  2 +-
 pkgs/applications/networking/cluster/nomad/default.nix                          | 12 ++++++------
 pkgs/applications/networking/cluster/octant/plugins/starboard-octant-plugin.nix |  2 +-
 pkgs/applications/networking/cluster/qbec/default.nix                           |  2 +-
 pkgs/applications/networking/cluster/sonobuoy/default.nix                       |  2 +-
 pkgs/applications/networking/cluster/temporalite/default.nix                    |  2 +-
 pkgs/applications/networking/cluster/terraform-docs/default.nix                 |  2 +-
 pkgs/applications/networking/cluster/terraform-inventory/default.nix            |  2 +-
 pkgs/applications/networking/cluster/tgswitch/default.nix                       |  2 +-
 pkgs/applications/networking/feedreaders/photon/default.nix                     |  2 +-
 pkgs/applications/networking/gmailctl/default.nix                               |  2 +-
 pkgs/applications/networking/hyprspace/default.nix                              |  2 +-
 pkgs/applications/networking/instant-messengers/go-neb/default.nix              |  2 +-
 pkgs/applications/networking/instant-messengers/gomuks/default.nix              |  2 +-
 pkgs/applications/networking/instant-messengers/mm/default.nix                  |  2 +-
 pkgs/applications/networking/irc/senpai/default.nix                             |  2 +-
 pkgs/applications/networking/p2p/magnetico/default.nix                          |  2 +-
 pkgs/applications/networking/peroxide/default.nix                               |  2 +-
 pkgs/applications/networking/r53-ddns/default.nix                               |  2 +-
 pkgs/applications/networking/stc-cli/default.nix                                |  2 +-
...skipping...
 pkgs/development/tools/database/timescaledb-parallel-copy/default.nix           |  2 +-
 pkgs/development/tools/database/timescaledb-tune/default.nix                    |  2 +-
 pkgs/development/tools/dstp/default.nix                                         |  2 +-
 pkgs/development/tools/easyjson/default.nix                                     |  2 +-
 pkgs/development/tools/ec2-metadata-mock/default.nix                            |  2 +-
 pkgs/development/tools/ejson/default.nix                                        |  2 +-
 pkgs/development/tools/ent/default.nix                                          |  2 +-
 pkgs/development/tools/esbuild/netlify.nix                                      |  2 +-
 pkgs/development/tools/faq/default.nix                                          |  2 +-
 pkgs/development/tools/garble/default.nix                                       |  2 +-
 pkgs/development/tools/github/bump/default.nix                                  |  2 +-
 pkgs/development/tools/go-junit-report/default.nix                              |  2 +-
 pkgs/development/tools/go-outline/default.nix                                   |  2 +-
 pkgs/development/tools/goda/default.nix                                         |  2 +-
 pkgs/development/tools/gofumpt/default.nix                                      |  2 +-
 pkgs/development/tools/gokart/default.nix                                       |  2 +-
 pkgs/development/tools/golines/default.nix                                      |  2 +-
 pkgs/development/tools/golint/default.nix                                       |  2 +-
 pkgs/development/tools/gomodifytags/default.nix                                 |  2 +-
 pkgs/development/tools/gotest/default.nix                                       |  2 +-
 pkgs/development/tools/gotests/default.nix                                      |  2 +-
 pkgs/development/tools/gotools/default.nix                                      |  2 +-
 pkgs/development/tools/gron/default.nix                                         |  2 +-
 pkgs/development/tools/hover/default.nix                                        |  2 +-
 pkgs/development/tools/htmltest/default.nix                                     |  2 +-
 pkgs/development/tools/ijq/default.nix                                          |  2 +-
 pkgs/development/tools/ineffassign/default.nix                                  |  2 +-
 pkgs/development/tools/jira-cli-go/default.nix                                  |  2 +-
 pkgs/development/tools/jmespath/default.nix                                     |  2 +-
 pkgs/development/tools/jp/default.nix                                           |  2 +-
 pkgs/development/tools/json2hcl/default.nix                                     |  2 +-
 pkgs/development/tools/kube-prompt/default.nix                                  |  2 +-
 pkgs/development/tools/kubectx/default.nix                                      |  2 +-
 pkgs/development/tools/kustomize/3.nix                                          |  2 +-
 pkgs/development/tools/language-servers/buf-language-server/default.nix         |  2 +-
 pkgs/development/tools/leaps/default.nix                                        |  2 +-
 pkgs/development/tools/maligned/default.nix                                     |  2 +-
 pkgs/development/tools/misc/aviator/default.nix                                 |  2 +-
 pkgs/development/tools/misc/jiq/default.nix                                     |  2 +-
 pkgs/development/tools/misc/k2tf/default.nix                                    |  2 +-
 pkgs/development/tools/misc/mkcert/default.nix                                  |  2 +-
 pkgs/development/tools/mockgen/default.nix                                      |  2 +-
 pkgs/development/tools/modd/default.nix                                         |  2 +-
 pkgs/development/tools/mustache-go/default.nix                                  |  2 +-
 pkgs/development/tools/nap/default.nix                                          |  2 +-
 pkgs/development/tools/nc4nix/default.nix                                       |  2 +-
 pkgs/development/tools/oshka/default.nix                                        |  2 +-
 pkgs/development/tools/out-of-tree/default.nix                                  |  2 +-
 pkgs/development/tools/pigeon/default.nix                                       |  2 +-
 pkgs/development/tools/proto-contrib/default.nix                                |  2 +-
 pkgs/development/tools/protoc-gen-doc/default.nix                               |  2 +-
 pkgs/development/tools/protoc-gen-twirp_swagger/default.nix                     |  2 +-
 pkgs/development/tools/protoc-gen-twirp_typescript/default.nix                  |  2 +-
 pkgs/development/tools/protolint/default.nix                                    |  2 +-
 pkgs/development/tools/rakkess/default.nix                                      |  2 +-
 pkgs/development/tools/reflex/default.nix                                       |  2 +-
 pkgs/development/tools/refmt/default.nix                                        |  2 +-
 pkgs/development/tools/spicy/default.nix                                        |  2 +-
 pkgs/development/tools/supabase-cli/default.nix                                 |  2 +-
 pkgs/development/tools/toxiproxy/default.nix                                    |  2 +-
 pkgs/development/tools/turbogit/default.nix                                     |  2 +-
 pkgs/development/tools/unconvert/default.nix                                    |  2 +-
 pkgs/development/tools/wails/default.nix                                        |  2 +-
 pkgs/development/tools/wally-cli/default.nix                                    |  2 +-
 pkgs/development/tools/wire/default.nix                                         |  2 +-
 pkgs/development/web/shopify-themekit/default.nix                               |  2 +-
 pkgs/games/itch/butler.nix                                                      |  2 +-
 pkgs/games/uchess/default.nix                                                   |  2 +-
 pkgs/os-specific/linux/cshatag/default.nix                                      |  2 +-
 pkgs/os-specific/linux/ipp-usb/default.nix                                      |  2 +-
 pkgs/os-specific/linux/pam_ussh/default.nix                                     |  2 +-
 pkgs/os-specific/linux/ultrablue-server/default.nix                             |  2 +-
 pkgs/servers/alice-lg/default.nix                                               |  2 +-
 pkgs/servers/alps/default.nix                                                   |  2 +-
 pkgs/servers/asouldocs/default.nix                                              |  2 +-
 pkgs/servers/bird-lg/default.nix                                                |  8 ++++----
 pkgs/servers/birdwatcher/default.nix                                            |  2 +-
 pkgs/servers/cayley/default.nix                                                 |  2 +-
 pkgs/servers/confluencepot/default.nix                                          |  2 +-
 pkgs/servers/dns/ncdns/default.nix                                              |  2 +-
 pkgs/servers/echoip/default.nix                                                 |  2 +-
 pkgs/servers/geospatial/mbtileserver/default.nix                                |  2 +-
 pkgs/servers/go-autoconfig/default.nix                                          |  2 +-
 pkgs/servers/go-cqhttp/default.nix                                              |  2 +-
 pkgs/servers/gonic/default.nix                                                  |  2 +-
 pkgs/servers/gotty/default.nix                                                  |  2 +-
 pkgs/servers/hiraeth/default.nix                                                |  2 +-
 pkgs/servers/http/pomerium/default.nix                                          |  2 +-
 pkgs/servers/http/ran/default.nix                                               |  2 +-
 pkgs/servers/interlock/default.nix                                              |  2 +-
 pkgs/servers/kubemq-community/default.nix                                       |  2 +-
 pkgs/servers/lenpaste/default.nix                                               |  2 +-
 pkgs/servers/livepeer/default.nix                                               |  2 +-
 pkgs/servers/maddy/default.nix                                                  |  2 +-
 pkgs/servers/mail/listmonk/default.nix                                          |  2 +-
 pkgs/servers/matrix-corporal/default.nix                                        |  2 +-
 pkgs/servers/mautrix-discord/default.nix                                        |  2 +-
 pkgs/servers/mautrix-whatsapp/default.nix                                       |  2 +-
 pkgs/servers/minio/legacy_fs.nix                                                |  2 +-
 pkgs/servers/misc/podgrab/default.nix                                           |  2 +-
 pkgs/servers/monitoring/prometheus/apcupsd-exporter.nix                         |  2 +-
 pkgs/servers/monitoring/prometheus/blackbox-exporter.nix                        |  2 +-
 pkgs/servers/monitoring/prometheus/cloudflare-exporter.nix                      |  2 +-
 pkgs/servers/monitoring/prometheus/flow-exporter.nix                            |  2 +-
 pkgs/servers/monitoring/prometheus/junos-czerwonk-exporter.nix                  |  2 +-
 pkgs/servers/monitoring/prometheus/keylight-exporter.nix                        |  2 +-
 pkgs/servers/monitoring/prometheus/modemmanager-exporter.nix                    |  2 +-
 pkgs/servers/monitoring/prometheus/nginx-exporter.nix                           |  2 +-
 pkgs/servers/monitoring/prometheus/postfix-exporter.nix                         |  2 +-
 pkgs/servers/monitoring/prometheus/process-exporter.nix                         |  2 +-
 pkgs/servers/monitoring/prometheus/promscale/default.nix                        |  2 +-
 pkgs/servers/monitoring/prometheus/shelly-exporter.nix                          |  2 +-
 pkgs/servers/monitoring/prometheus/systemd-exporter.nix                         |  2 +-
 pkgs/servers/monitoring/prometheus/v2ray-exporter.nix                           |  2 +-
 pkgs/servers/monitoring/zabbix/agent2.nix                                       |  4 ++--
 pkgs/servers/monitoring/zabbix/versions.nix                                     |  2 +-
 pkgs/servers/nosql/ferretdb/default.nix                                         |  2 +-
 pkgs/servers/nosql/influxdb/default.nix                                         |  2 +-
 pkgs/servers/nosql/influxdb2/default.nix                                        |  2 +-
 pkgs/servers/oauth2-proxy/default.nix                                           |  2 +-
 pkgs/servers/photoprism/backend.nix                                             |  2 +-
 pkgs/servers/rmfakecloud/default.nix                                            |  2 +-
 pkgs/servers/serf/default.nix                                                   |  2 +-
 pkgs/servers/sql/rqlite/default.nix                                             |  2 +-
 pkgs/servers/swego/default.nix                                                  |  2 +-
 pkgs/servers/tracing/honeycomb/honeymarker/default.nix                          |  2 +-
 pkgs/servers/tracing/honeycomb/honeytail/default.nix                            |  2 +-
 pkgs/servers/trezord/default.nix                                                |  2 +-
 pkgs/servers/unifiedpush-common-proxies/default.nix                             |  2 +-
 pkgs/servers/web-apps/morty/default.nix                                         |  2 +-
 pkgs/servers/web-apps/vikunja/api.nix                                           |  2 +-
 pkgs/servers/webdav/default.nix                                                 |  2 +-
 pkgs/servers/wesher/default.nix                                                 |  2 +-
 pkgs/servers/xteve/default.nix                                                  |  2 +-
 pkgs/shells/fish/babelfish.nix                                                  |  2 +-
 pkgs/shells/oh/default.nix                                                      |  2 +-
 pkgs/shells/zsh/zsh-history/default.nix                                         |  2 +-
 pkgs/tools/X11/go-sct/default.nix                                               |  2 +-
 pkgs/tools/admin/awsls/default.nix                                              |  2 +-
```

Double checking with a second nixpkgs-review run

**Comment by fabianhjr** [comment]
1 package removed:
zabbix-agent2 (†6.0.14)

**Comment by fabianhjr** [comment]
Yay, zero rebuilds and no longer gets the cleanup label

**Review by Mindavi — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Pull request merged**